### ISSUE TITLE
Can't run mapshaper-xl commands programmatically

### ISSUE THREAD

**Issue by davidvermeir** [mbloch/mapshaper#367]
Mapshaper has what I can consider to be a great feature where you can run the commands programmatically directly from Node:

`var mapshaper = require('mapshaper');
 mapshaper.runCommands('-i shapefiles/*.shp -o geojson/ format=geojson');`

However, I've noticed that it isn't possible to run mapshaper-xl commands in the same way. For triggering parsing of large files this would be very useful.

Unless I'm overlooking something?

**Comment by mbloch** [comment]
This is a good suggestion... I'll think about how to support allocating more memory to the `runCommands()` function.

Regarding the API, an optional second argument to`runCommands()` might work:

`mapshaper.runCommands(commands, {xl: true})`
`mapshaper.runCommands(commands, {xl: "16gb"})`

**Comment by mbloch** [comment]
I just published a new release with this functionality.

Instead of adding an option to `runCommands()`, I added a new API function, `runCommandsXL()`, which takes an optional value for setting a custom value for the heap, e.g. `{xl: "16gb"}`. The default heap size of `runCommandsXL()` is the same as `mapshaper-xl`: 8gb.

I'll document this in the mapshaper wiki soon.

The reason for adding a new command instead of overloading `runCommands()` is that setting a custom heap size would change the behavior of `runCommands()` a bit. To change the default heap size, I have to run Node + mapshaper as a subprocess, which means it's no longer possible to pass in input datasets as an argument to the `runCommands()` function. `runCommandsXL()` is limited to file i/o.
